datadoc/tests/test_sas.py

```diff
@@ -0,0 +1,46 @@
+# TODO: Could this be useful?
+# sasdata = sas_session.sasdata("sasdata")
+# columnInfo = sasdata.columnInfo()
+
+
+import saspy
+from pathlib import PurePath
+from .utils import TEST_SAS_FILEPATH, TEST_SAS_PROGRAM
+
+
+def setup_module():
+    global sas_session
+    sas_session = saspy.SASsession()
+
+
+def run_sas_program(sas_session):
+    code = open(PurePath(TEST_SAS_PROGRAM)).read()
+    return sas_session.submit(code)
+
+
+def download_sas_dataset(sas_session):
+    remote_file = sas_session.workpath + PurePath(TEST_SAS_FILEPATH).name
+    return sas_session.download(PurePath(TEST_SAS_FILEPATH), remote_file)
+
+
+def test_submit_noerrorinlog():
+    run_log = run_sas_program(sas_session)
+    error = False
+    if 'ERROR' in run_log['LOG']:
+        error = True
+    assert error is False
+
+
+def test_download_success():
```

Comment:
These functions look really useful for creating and updating our SAS test data! However I feel like these functions would be better suited as standalone utilities rather than tests. The concerns I have are:

- They aren't testing the main application and as such, a failure of one of these tests isn't indicative of a problem with Datadoc itself
- These tests are dependent on specific local configuration (`sascfg_personal.py`), which may not be present in all environments (for example Github Actions). When running the test locally I get `saspy.sasexceptions.SASIOConnectionError: Failure establishing SASsession.
 No SAS process attached. SAS process has terminated unexpectedly.`
- These tests depend entirely on an external service and are therefore likely to be flaky